tidb-in-kubernetes/reference/components/tidb-scheduler.md

```diff
@@ -0,0 +1,60 @@
+---
+title: TiDB Scheduler
+category: components
+Summary: Learn how TiDB Scheduler works
+---
+
+# TiDB Scheduler
+
+This document introduces how TiDB Scheduler works.
+
+## TiDB cluster scheduling requirements
+
+The TiDB cluster includes three key components: PD, TiKV, and TiDB. Each consists of multiple nodes: PD is a Raft cluster, and TiKV is a multi-group Raft cluster. Both components are stateful. Therefore, the default scheduling rules of the Kubernetes (K8s) scheduler can no longer meet the TiDB cluster scheduling requirements. To extend the K8s scheduling rules, the TiDB Operator implements the following customized scheduling rules:
+
+### PD
+
+Make sure that the number of PDs scheduled on each Node is less than `Replicas / 2`, for example:
+
+| PD cluster size (Replicas) | Maximum number of PDs scheduled on each node |
+| ------------- | ------------- |
+| 1  | 1  |
+| 2  | 1  |
+| 3  | 1  |
+| 4  | 1  |
+| 5  | 2  |
+| ...  |   |
+
+### TiKV
+
+If the number of K8s nodes is less than three (In this case, TiKV is not highly available), arbitrary schedule is supported; otherwise, the formula for calculating the number of TiKV schedulable on each node is: Ceil (Replicas / 3), for example:
+
+| TiKV cluster size (Replicas) | Maximum number of TiKVs scheduled on each node | Best scheduling distribution |
+| ------------- | ------------- | ------------- |
+| 3  | 1  | 1,1,1  |
+| 4  | 2  | 1,1,2  |
+| 5  | 2  | 1,2,2  |
+| 6  | 2  | 2,2,2  |
+| 7  | 3  | 2,2,3  |
+| 8  | 3  | 2,3,3  |
+| ...  |   |   |
+
+### TiDB
+
+Stable scheduling is achieved. In the case of TiDB rolling upgrade, it tends to be scheduled back to the original nodes. This is helpful for the scenario of manually mounting Node IP and NodePort on the LB backend. It avoids rearranging the LB when the Node IP is changed after the upgrade, thereby reducing the impact on the cluster during rolling upgrade.
```

Comment:
```suggestion
Scheduling rule 3: When you perform a rolling upgrade to a TiDB instance, the instance tends to be scheduled back to its original node. 

This ensures stable scheduling and is helpful for the scenario of manually mounting Node IP and NodePort to the LB backend. It can reduce the impact on the cluster during the rolling upgrade because you do not need to rearrange the LB when the Node IP is changed after the upgrade.
```